test/autorest/generated/lrogroup/pollers.go

```diff
@@ -193,6 +195,14 @@ func productPollerPollUntilDone(ctx context.Context, p ProductPoller, frequency
 	return p.FinalResponse(ctx)
 }
 
+func productPollerHandleResponse(resp *azcore.Response) (*ProductResponse, error) {
+	result := ProductResponse{RawResponse: resp.Response}
+	if resp.HasStatusCode(http.StatusNoContent) {
+		return &result, nil
+	}
+	return &result, resp.UnmarshalAsJSON(&result.Product)
```

Comment:
This will end up unmarshalling on a non-successful status code which will likely fail as there's no response body.  At the least I'm thinking we should ensure that the status code here is `pollingCodes`, what do you think?

Reply:
Mmm, I think there is a possibility that that could happen. I'll add a check with `pollingCodes` I think that's a good idea.

Reply:
So I had to add a handleError method on pollingTrackerBase. If you can go look at that change.